### PR TITLE
Add rudimentary glossary

### LINKED ISSUE #36: Create a glossary to document jargon, terms, and definitions
There are bunch of terms that get used in the code of x-api/trackler/exercism.io/configlet etc.

Such as: Track, Exercise, Slug, ID, Problem, Specification, Implementation

It would be good to have a place where we could canonically list the definitions, and their intended usage.

### PR DESCRIPTION
**Pull request by kytrinyx**
I'm sure we'll find more terms to add to this, but I think this is a decent first pass.

Fixes #36

### CODE REVIEW

**Review by Insti — APPROVED** [review]

File: glossary.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Glossary
+
+There are a number of terms that we use that have specific meaning within the
+context of Exercism.
+
+- **language** - the name of a programming language, for example _C++_ or _Haskell_
+- **track** - a collection of exercises implemented in the a given programming language
+- **track id** - a short, unique identifier for a track (e.g. `cpp` or `haskell`). Matches `/^[a-z-]+$/`
+- **problem** - abstract concept for a programming task on exercism.io
+- **specification** - a language-agnostic description of a problem. These live in [the x-common repository][x-common-repo]
```
Comment (Insti) [inline glossary.md]: Probably the wrong place for this thought but: 
I wonder if renaming the `exercises` directory of x-common to `specifications` is a good idea.

Comment (kytrinyx) [reply glossary.md]: ooooh, that is an excellent idea. Would you open an issue in the discussions repo for it?

Comment (Insti) [reply glossary.md]: I'll wait until the glossary exists then I can refer to it.

File: glossary.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Glossary
+
+There are a number of terms that we use that have specific meaning within the
+context of Exercism.
+
+- **language** - the name of a programming language, for example _C++_ or _Haskell_
+- **track** - a collection of exercises implemented in the a given programming language
```
Comment (ErikSchierboom) [inline glossary.md]: In the a => in a

**Review by ErikSchierboom — CHANGES_REQUESTED** [review]

File: glossary.md (hunk 1 above)
Comment (ErikSchierboom) [reply glossary.md]: Great idea!

File: glossary.md (hunk 2 above)
Comment (kytrinyx) [reply glossary.md]: Fixed!

**Review by ErikSchierboom — APPROVED** [review]

File: glossary.md (hunk 1 above)
Comment (Insti) [reply glossary.md]: https://github.com/exercism/discussions/issues/145

**Pull request merged**